GitUI/CommandsDialogs/FormPush.cs

```diff
@@ -502,15 +502,15 @@ private bool HandlePushOnExit(ref bool isError, FormProcess form)
 
                 if (forcePush)
                 {
-                    if (!form.ProcessArguments.Contains(" -f ") && !form.ProcessArguments.Contains(" --force"))
+                    if (!form.ProcessArguments.Contains(" -f ") && !form.ProcessArguments.Contains(" --force") && !form.ProcessArguments.EndsWith(" -f"))
```

Comment:
I don't understand why you're checking if the args ends with -f here. Even if it were necessary, wouldn't you want to also check if it's --force?